lib/virtfs/xfs/file_class.rb

```diff
@@ -0,0 +1,224 @@
+module VirtFS::XFS
+  class FS
```

Comment:
@Fryguy Actually, I tried the mixin route initially and I found it to be more cumbersome overall.
Also, I recall I may have encountered some technical snags as well. But that may have been in the core VirtFS code - which has the same issue.

I'm not opposed to the change. FS modules are far simpler than the core VirtFS, so it may not be an issue. There are three areas that might be problematic: the core VirtFS, the NativeFS (thick vs thin) and CamcorderFS (which is the most complex FS module).

So, I say we go with the mixin's for the simple FS modules, and then revisit the more complex cases to see if we can address those as well.